config/validate/node.go

```diff
@@ -94,15 +94,30 @@ func toNode(v interface{}, c context, n *node) {
 		}
 	case reflect.Map:
 		// Walk over each key in the map and create a node for it.
-		v := v.(map[interface{}]interface{})
-		for k, cv := range v {
-			cn := node{name: fmt.Sprintf("%s", k)}
-			c, ok := findKey(cn.name, c)
-			if ok {
-				cn.line = c.lineNumber
+		cpv := v
+		v, ok := cpv.(map[interface{}]interface{})
+		if ok {
+			for k, cv := range v {
+				cn := node{name: fmt.Sprintf("%s", k)}
+				c, ok := findKey(cn.name, c)
+				if ok {
+					cn.line = c.lineNumber
+				}
+				toNode(cv, c, &cn)
+				n.children = append(n.children, cn)
 			}
-			toNode(cv, c, &cn)
-			n.children = append(n.children, cn)
+		} else {
+			sv := cpv.(map[string]interface{})
+			for k, cv := range sv {
+				cn := node{name: fmt.Sprintf("%s", k)}
+				c, ok := findKey(cn.name, c)
+				if ok {
+					cn.line = c.lineNumber
+				}
+				toNode(cv, c, &cn)
+				n.children = append(n.children, cn)
+			}
+
```

Comment:
You probably could avoid this duplication by putting the common code into an inline function (excuse my rusty go, haven't typed that in a long time):

```go
handleKV := func(k,v interface{}) {
	cn := node{name: fmt.Sprintf("%s", k)}
	c, ok := findKey(cn.name, c)
	if ok {
		cn.line = c.lineNumber
	}
	toNode(cv, c, &cn)
	n.children = append(n.children, cn)
}
cpv := v
v, ok := cpv.(map[interface{}]interface{})
if ok {
	for k, cv := range v {
		handleKV(k, cv)
	}
} else {
	sv := cpv.(map[string]interface{})
	for k, cv := range sv {
		handleKV(k, cv)
	}
}
```

Probably could be made even shorter by using Go's generics, but I have no experience with those.